Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -222,16 +222,18 @@
 % innerRepetition: single-volume repetitions for each PLD
 
 % Calculate the number of inner repetitions.
-% Let's take an example in which our volumes are stored as PLDs 1000, 1000, 1500, 1500, 2000, 2000 (ms)
-% That is nInnerRepetitions == 2, e.g., 2 echoes in the case of multi-TE.
+% Let's take an example in which our volumes are stored as PLDs 
+% 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1500, 1500, 1500, 1500, 1500, 1500, 1500, 1500, 
+% 2000, 2000, 2000, 2000, 2000, 2000, 2000, 2000, 1000, 1000, etc.
+% That is nInnerRepetitions == 8, e.g., 8 echoes in the case of multi-TE.
 
 [~, ~, indexUniquePLD] = unique(xQ.Initial_PLD); % Find volume indices of unique PLDs
 indexSecondPLD = find(indexUniquePLD==2); % Find the volume indices of the second PLD (1500). 
-% PLDs 1500 are on the position 3 and 4. And it means that all PLDs before the 1500 are equivalent (== PLD 1000 ms), assuming that 
-% all PLDs are incrementally stored
+% PLDs 1500 are on the position 9 and 10. And it means that all PLDs before the 1500 are equivalent (== PLD 1000 ms), because PLDs==1500 has been labeled as the second different value by unique
+
 if isempty(indexSecondPLD)
 	nInnerRepetitions = length(xQ.Initial_PLD);
-    % If each PLD has only a single volume (and no second volume), then nInnerRepetitions is the length of the PLD vector
+    % In the unlikely event that there is a single unique PLD value only, then nInnerRepetitions is the length of the PLD vector
```

Comment:
`nInnerRepetitions = indexSecondPLD(1) - 1;`

Note that this gives me 16 instead of 8, for the HAD-4 example with 8 echoes and 2 nOuterRepetitions.

Did you test this already?

Reply:
See whatsap. tested in 1000, 1500, 2500, where it worked. 2500, 1000, 1500 indeed doesn't. Nice catch. Fixed in the last commit.